tests/unit/io/UAIFile_unittest.cc

```diff
@@ -67,7 +67,7 @@ TEST(UAIFileTest, preamble)
                       factors_scope_in);
 
     const char net_type_expected[] = "MARKOV";
-    EXPECT_EQ(net_type.vlen, strlen(net_type_expected));
+    EXPECT_EQ(net_type.vlen, static_cast<int32_t>(strlen(net_type_expected)));
```

Comment:
Minor: Technically, this is not safe. Think of possible overflows.
You could rather cast the vlen to unsigned int and assert for positivity first.